R/extractVariableGroups.R

```diff
@@ -3,20 +3,27 @@
 #' Groups variable names by groups based on the |+| separators given in the variable names
 #' 
 #' @param x a vector of variable names
+#' @param keepOrigNames if set, the returned list contains the original variables
+#' (to the value of which the grouped ones have to sum up) as names instead of
+#' made up group names, if they exist. The current implementation goes up to two levels (++) deep.
 #' @return a named list of variable groups with group name as name and vector of entities as content
 #' @author David Klein, Jan Philipp Dietrich
 #' @seealso \code{\link{plotstyle.add}}
 #' @examples
 #' x <- c("a|+|1|+|aa","a|+|2|abc","a|+|1|+|bb","a|+|1|+|cc","a|+|3|+|aa","a|+|3|+|bb")
-#' mip:::extractVariableGroups(x)
+#' mip::extractVariableGroups(x)
+#' @export
 
-extractVariableGroups <- function(x) {
-  tmp <- function(x,sep="|+|",ext="") {
+extractVariableGroups <- function(x,keepOrigNames=FALSE) {
+  tmp <- function(x,sep="|+|",ext="",allVars) {
     y <- strsplit(x,sep, fixed=TRUE)
     out <- list()
     for(j in 1:length(y)) {
       for(i in 1:(length(y[[j]])-1)) {
         name <- paste0(paste(y[[j]][1:i],collapse=sep),ext)
+        ind<-grep(gsub("\\|","\\\\|",paste0("^",name,"$")),
+                  sub("\\|\\+\\+|\\|\\+","",sub(" \\(.*.\\)$","",allVars)))
+        if (keepOrigNames ) try(name<-allVars[[ind]],silent = T)
```

Comment:
keepOrigNames is missing in the argument list of tmp

Reply:
I added github actions and get this error: https://github.com/giannou/madrat/runs/1934783668?check_suite_focus=true
Any idea what the problem might be?